dataplane/linux/vxlan_mgr.go

```diff
@@ -59,47 +74,83 @@ type vxlanManager struct {
 	ipSetMetadata     ipsets.IPSetMetadata
 	externalNodeCIDRs []string
 	vtepsDirty        bool
+	nlHandle          vxlanDataplane
+	dpConfig          Config
+	parentProtocol    int
+	// Used so that we can shim the parent route table for the tests
+	parentRTConstruct func(interfacePrefixes []string, ipVersion uint8, vxlan bool, netlinkTimeout time.Duration,
+		deviceRouteSourceAddress net.IP, deviceRouteProtocol int, removeExternalRoutes bool) routeTable
 }
 
 func newVXLANManager(
 	ipsetsDataplane ipsetsDataplane,
-	maxIPSetSize int,
-	hostname string,
-	routeTable routeTable,
+	rt routeTable,
+	deviceName string,
+	dpConfig Config,
+) *vxlanManager {
+	nlHandle, _ := netlink.NewHandle()
+
+	return newVXLANManagerWithShims(
+		ipsetsDataplane,
+		rt,
+		deviceName,
+		dpConfig,
+		nlHandle,
+		func(interfacePrefixes []string, ipVersion uint8, vxlan bool, netlinkTimeout time.Duration,
+			deviceRouteSourceAddress net.IP, deviceRouteProtocol int, removeExternalRoutes bool) routeTable {
+			return routetable.New(interfacePrefixes, ipVersion, vxlan, netlinkTimeout,
+				deviceRouteSourceAddress, deviceRouteProtocol, removeExternalRoutes)
+		},
+	)
+}
+
+func newVXLANManagerWithShims(
+	ipsetsDataplane ipsetsDataplane,
```

Comment:
So, do we use this `ipsetsDataplane` anywhere? Is this just a remnant from when you were looking at using ipsets here?

Reply:
line 252, m.ipsetsDataplane.AddOrReplaceIPSet(m.ipSetMetadata, allowedVXLANSources)